### ISSUE TITLE
[Mismatch of Codes] built libtensorflow_cc.so does not match the interface of Eigen 

### ISSUE THREAD

**Issue by yiakwy-enterprise-roborock** [tensorflow/tensorflow#38237]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): source
- TensorFlow version: 2.2.0
- Python version: python3.6 （maintained by conda）
- Installed using virtualenv? pip? conda?: conda
- Bazel version (if compiling from source): 0.29.1 (according to requirements of tensorflow 2.2.0)
- GCC/Compiler version (if compiling from source): 7.5.0
- CUDA/cuDNN version: 10.1
- GPU model and memory: ON 7 GB

**Describe the problem**

**Provide the exact sequence of commands / steps that you executed before running into the problem**

I built tensorflow from sources and exported both dependancies of files and the shared library to /usr/local/include/tensorflow, /usr/local/lib (soft link to /usr/lib/ ...)

Now I am building a cmake projects using tensorflow. Here is a cmake snopshot:

```txt
 cmake ..
[Semantic_Relocalization] Build type: Release
[Semantic_Relocalization] Install path: /usr/local
-- Using libstdc++(gcc). You can change the tool chain with clang/clang++ in Ubuntu/CentOS/MacOS
-- Could NOT find GTest (missing: GTEST_LIBRARY GTEST_MAIN_LIBRARY)
-- Found gflags (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libgflags.so)
-- GFLAGS_INCLUDE_DIRS: /usr/include/gflags
-- GFLAGS_LIBRARIES: /usr/lib/x86_64-linux-gnu/libgflags.so
-- Found glog (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libglog.so)
-- GLOG_INCLUDE_DIRS: /usr/include/glog
-- GLOG_LIBRARIES: /usr/lib/x86_64-linux-gnu/libglog.so
-- Home directory : /home/yiakwy
CMake Warning at cmake/Relies.cmake:17 (message):
  System variable EXTERNAL_DIR is nonexist, using
  /home/yiakwy/mapping_external instead.
Call Stack (most recent call first):
  CMakeLists.txt:123 (include)


-- EXTERNAL_LIBS_DIR : /home/yiakwy/mapping_external/linux
-- Find Opencv (ver.) (include: /usr/local/include/opencv4, library: opencv_calib3d
-- opencv_core
-- opencv_dnn
-- opencv_features2d
-- opencv_flann
-- opencv_gapi
-- opencv_highgui
-- opencv_imgcodecs
-- opencv_imgproc
-- opencv_ml
-- opencv_objdetect
-- opencv_photo
-- opencv_stitching
-- opencv_video
-- opencv_videoio
-- opencv_aruco
-- opencv_bgsegm
-- opencv_bioinspired
-- opencv_ccalib)
CMake Warning at /usr/local/share/cmake-3.9/Modules/FindBoost.cmake:769 (message):
  Imported targets not available for Boost version 106501
Call Stack (most recent call first):
  /usr/local/share/cmake-3.9/Modules/FindBoost.cmake:873 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/share/cmake-3.9/Modules/FindBoost.cmake:1501 (_Boost_MISSING_DEPENDENCIES)
  cmake/Relies.cmake:58 (find_package)
  CMakeLists.txt:123 (include)


CMake Warning at /usr/local/share/cmake-3.9/Modules/FindBoost.cmake:769 (message):
  Imported targets not available for Boost version 106501
Call Stack (most recent call first):
  /usr/local/share/cmake-3.9/Modules/FindBoost.cmake:873 (_Boost_COMPONENT_DEPENDENCIES)
  /usr/local/share/cmake-3.9/Modules/FindBoost.cmake:1501 (_Boost_MISSING_DEPENDENCIES)
  cmake/Relies.cmake:58 (find_package)
  CMakeLists.txt:123 (include)


-- Boost version: 1.65.1
-- Found the following Boost libraries:
--   system
--   filesystem
-- EIGEN3_VERSION: 3.3.90
-- Eigen3_DIR: /usr/local/share/eigen3/cmake
-- Found Eigen3, inc: /usr/local/include/eigen3
-- Found Eigen3: inc /usr/local/include/eigen3 (ver: 3.3.90)
-- Protobuf compiler version: libprotoc 3.8.0

-- Found Protobuf: /usr/local/lib/libprotobuf.so;-lpthread (found version "3.8.0")
-- Protobuf compiler version: libprotoc 3.8.0

-- Found Protobuf: /usr/local/lib/libprotobuf.so;-lpthread;-lpthread (found version "3.8.0")
-- Using protobuf 3.8.0
-- Protobuf_PROTOC_EXECUTABLE: /usr/local/bin/protoc
-- Found Tensorflow, inc: /usr/local/include/google/tensorflow, lib: /usr/local/lib/libtensorflow_all.so
-- PROTO_FILES: /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/proto/runtime_block.proto
-- hw_proto_srcs: /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/build/proto_codec/runtime_block.pb.cc
-- hw_proto_hdrs: /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/build/proto_codec/runtime_block.pb.h
-- PROTOBUF_INCLUDE_DIRS: /usr/local/include
-- PROTOBUF_LIBRARIES: /usr/local/lib/libprotobuf.so
-- -lpthread
-- -lpthread
-- models PYTHON_INCLUDE_DIR: /home/yiakwy/anaconda3/envs/py36/include/python3.6m
-- models PYTHON_LIBRARY: /home/yiakwy/anaconda3/envs/py36/lib/libpython3.6m.so.1.0
-- PYTHONHOME: /home/yiakwy/anaconda3/envs/py36
-- Configuring done
-- Generating done
-- Build files have been written to: /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/build

```

However when I try to make it, something goes wrong with Eigen interface:

```text
▶ make
[ 50%] Built target base
[ 62%] Building CXX object modules/models/CMakeFiles/models.dir/main.cpp.o
In file included from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/FixedPoint:42:0,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/numeric_types.h:24,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/allocator.h:26,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:22,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/MatMatProductAVX2.h:49:76: error: wrong number of template arguments (5, should be at least 3)
 class TensorContractionBlocking<QInt16, QInt16, QInt16, Index, ShardingType> {
                                                                            ^
In file included from /usr/include/eigen3/unsupported/Eigen/CXX11/Tensor:112:0,
                 from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:21,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorContractionBlocking.h:25:7: note: provided for ‘template<class LhsMapper, class RhsMapper, class Index, int ShardingType> class Eigen::internal::TensorContractionBlocking’
 class TensorContractionBlocking {
       ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/FixedPoint:42:0,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/numeric_types.h:24,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/allocator.h:26,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:22,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/MatMatProductAVX2.h:151:42: error: wrong number of template arguments (9, should be at least 6)
                      Conjugate, PanelMode> {
                                          ^
In file included from /usr/include/eigen3/unsupported/Eigen/CXX11/../../../Eigen/Core:429:0,
                 from /usr/include/eigen3/unsupported/Eigen/CXX11/Tensor:14,
                 from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:21,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/include/eigen3/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/BlasUtil.h:28:8: note: provided for ‘template<class Scalar, class Index, class DataMapper, int Pack1, int Pack2, int StorageOrder, bool Conjugate, bool PanelMode> struct Eigen::internal::gemm_pack_lhs’
 struct gemm_pack_lhs;
        ^~~~~~~~~~~~~
In file included from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/FixedPoint:42:0,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/numeric_types.h:24,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/allocator.h:26,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:22,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/MatMatProductAVX2.h:160:76: error: wrong number of template arguments (9, should be at least 6)
                                      QInt16, ColMajor, Conjugate, PanelMode>::
                                                                            ^
In file included from /usr/include/eigen3/unsupported/Eigen/CXX11/../../../Eigen/Core:429:0,
                 from /usr/include/eigen3/unsupported/Eigen/CXX11/Tensor:14,
                 from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:21,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/include/eigen3/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/BlasUtil.h:28:8: note: provided for ‘template<class Scalar, class Index, class DataMapper, int Pack1, int Pack2, int StorageOrder, bool Conjugate, bool PanelMode> struct Eigen::internal::gemm_pack_lhs’
 struct gemm_pack_lhs;
        ^~~~~~~~~~~~~
In file included from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/FixedPoint:42:0,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/numeric_types.h:24,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/allocator.h:26,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:22,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/MatMatProductAVX2.h:162:38: error: ‘void Eigen::internal::operator()(Eigen::QInt16*, const DataMapper&, Index, Index, Index, Index)’ must be a nonstatic member function
            Index stride, Index offset) {
                                      ^
/usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/MatMatProductAVX2.h:549:24: error: wrong number of template arguments (5, should be at least 3)
     Index, ShardingType> {
                        ^
In file included from /usr/include/eigen3/unsupported/Eigen/CXX11/Tensor:112:0,
                 from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:21,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorContractionBlocking.h:25:7: note: provided for ‘template<class LhsMapper, class RhsMapper, class Index, int ShardingType> class Eigen::internal::TensorContractionBlocking’
 class TensorContractionBlocking {
       ^~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/FixedPoint:42:0,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/numeric_types.h:24,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/allocator.h:26,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:22,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/MatMatProductAVX2.h:1793:42: error: wrong number of template arguments (9, should be at least 6)
                      Conjugate, PanelMode> {
                                          ^
In file included from /usr/include/eigen3/unsupported/Eigen/CXX11/../../../Eigen/Core:429:0,
                 from /usr/include/eigen3/unsupported/Eigen/CXX11/Tensor:14,
                 from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:21,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/include/eigen3/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/BlasUtil.h:28:8: note: provided for ‘template<class Scalar, class Index, class DataMapper, int Pack1, int Pack2, int StorageOrder, bool Conjugate, bool PanelMode> struct Eigen::internal::gemm_pack_lhs’
 struct gemm_pack_lhs;
        ^~~~~~~~~~~~~
In file included from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/FixedPoint:42:0,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/numeric_types.h:24,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/allocator.h:26,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:22,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/MatMatProductAVX2.h:1802:75: error: wrong number of template arguments (9, should be at least 6)
                                      QInt8, ColMajor, Conjugate, PanelMode>::
                                                                           ^
In file included from /usr/include/eigen3/unsupported/Eigen/CXX11/../../../Eigen/Core:429:0,
                 from /usr/include/eigen3/unsupported/Eigen/CXX11/Tensor:14,
                 from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:21,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/include/eigen3/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/BlasUtil.h:28:8: note: provided for ‘template<class Scalar, class Index, class DataMapper, int Pack1, int Pack2, int StorageOrder, bool Conjugate, bool PanelMode> struct Eigen::internal::gemm_pack_lhs’
 struct gemm_pack_lhs;
        ^~~~~~~~~~~~~
In file included from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/FixedPoint:42:0,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/numeric_types.h:24,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/allocator.h:26,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:22,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/MatMatProductAVX2.h:1804:38: error: ‘void Eigen::internal::operator()(Eigen::QInt8*, const DataMapper&, Index, Index, Index, Index)’ must be a nonstatic member function
            Index stride, Index offset) {
                                      ^
In file included from /usr/include/eigen3/unsupported/Eigen/CXX11/Tensor:105:0,
                 from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:21,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorEvaluator.h: In instantiation of ‘struct Eigen::TensorEvaluator<const Eigen::TensorMap<Eigen::TensorFixedSize<float, Eigen::Sizes<>, 1, long int>, 16, Eigen::MakePointer>, Eigen::DefaultDevice>’:
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorForcedEval.h:96:65:   required from ‘struct Eigen::TensorEvaluator<const Eigen::TensorForcedEvalOp<const Eigen::TensorMap<Eigen::TensorFixedSize<float, Eigen::Sizes<>, 1, long int>, 16, Eigen::MakePointer>, Eigen::MakePointer>, Eigen::DefaultDevice>’
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorIO.h:61:42:   required from ‘std::ostream& Eigen::operator<<(std::ostream&, const Eigen::TensorBase<Derived, 0>&) [with T = Eigen::TensorMap<Eigen::TensorFixedSize<float, Eigen::Sizes<>, 1, long int>, 16, Eigen::MakePointer>; std::ostream = std::basic_ostream<char>]’
/home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:48:54:   required from here
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorEvaluator.h:162:71: warning: ignoring attributes on template argument ‘Eigen::PacketType<float, Eigen::DefaultDevice>::type {aka __vector(8) float}’ [-Wignored-attributes]
     PacketAccess = (internal::unpacket_traits<PacketReturnType>::size > 1),
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~
In file included from /usr/include/eigen3/unsupported/Eigen/CXX11/Tensor:133:0,
                 from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:21,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorForcedEval.h: In instantiation of ‘const int Eigen::TensorEvaluator<const Eigen::TensorForcedEvalOp<const Eigen::TensorMap<Eigen::TensorFixedSize<float, Eigen::Sizes<>, 1, long int>, 16, Eigen::MakePointer>, Eigen::MakePointer>, Eigen::DefaultDevice>::PacketSize’:
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorForcedEval.h:104:32:   required from ‘struct Eigen::TensorEvaluator<const Eigen::TensorForcedEvalOp<const Eigen::TensorMap<Eigen::TensorFixedSize<float, Eigen::Sizes<>, 1, long int>, 16, Eigen::MakePointer>, Eigen::MakePointer>, Eigen::DefaultDevice>’
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorIO.h:61:42:   required from ‘std::ostream& Eigen::operator<<(std::ostream&, const Eigen::TensorBase<Derived, 0>&) [with T = Eigen::TensorMap<Eigen::TensorFixedSize<float, Eigen::Sizes<>, 1, long int>, 16, Eigen::MakePointer>; std::ostream = std::basic_ostream<char>]’
/home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:48:54:   required from here
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorForcedEval.h:100:20: warning: ignoring attributes on template argument ‘Eigen::PacketType<float, Eigen::DefaultDevice>::type {aka __vector(8) float}’ [-Wignored-attributes]
   static const int PacketSize = internal::unpacket_traits<PacketReturnType>::size;
                    ^~~~~~~~~~
In file included from /usr/include/eigen3/unsupported/Eigen/CXX11/Tensor:139:0,
                 from /usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1,
                 from /usr/local/include/google/tensorflow/tensorflow/core/framework/tensor.h:21,
                 from /usr/local/include/google/tensorflow/tensorflow/core/public/session.h:24,
                 from /home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:6:
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorExecutor.h: In instantiation of ‘static void Eigen::internal::TensorExecutor<Expression, Eigen::DefaultDevice, true>::run(const Expression&, const Eigen::DefaultDevice&) [with Expression = const Eigen::TensorEvalToOp<const Eigen::TensorMap<Eigen::TensorFixedSize<float, Eigen::Sizes<>, 1, long int>, 16, Eigen::MakePointer>, Eigen::MakePointer>]’:
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorForcedEval.h:128:109:   required from ‘bool Eigen::TensorEvaluator<const Eigen::TensorForcedEvalOp<ArgType, MakePointer_>, Device>::evalSubExprsIfNeeded(Eigen::TensorEvaluator<const Eigen::TensorForcedEvalOp<ArgType, MakePointer_>, Device>::CoeffReturnType*) [with ArgType = const Eigen::TensorMap<Eigen::TensorFixedSize<float, Eigen::Sizes<>, 1, long int>, 16, Eigen::MakePointer>; Device = Eigen::DefaultDevice; MakePointer_ = Eigen::MakePointer; Eigen::TensorEvaluator<const Eigen::TensorForcedEvalOp<ArgType, MakePointer_>, Device>::CoeffReturnType = float]’
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorIO.h:66:3:   required from ‘std::ostream& Eigen::operator<<(std::ostream&, const Eigen::TensorBase<Derived, 0>&) [with T = Eigen::TensorMap<Eigen::TensorFixedSize<float, Eigen::Sizes<>, 1, long int>, 16, Eigen::MakePointer>; std::ostream = std::basic_ostream<char>]’
/home/yiakwy/WorkSpace/Github/SEMANTIC_SLAM/modules/models/main.cpp:48:54:   required from here
/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorExecutor.h:61:17: warning: ignoring attributes on template argument ‘Eigen::TensorEvaluator<const Eigen::TensorEvalToOp<const Eigen::TensorMap<Eigen::TensorFixedSize<float, Eigen::Sizes<>, 1, long int>, 16, Eigen::MakePointer>, Eigen::MakePointer>, Eigen::DefaultDevice>::PacketReturnType {aka __vector(8) float}’ [-Wignored-attributes]
       const int PacketSize = unpacket_traits<typename TensorEvaluator<Expression, DefaultDevice>::PacketReturnType>::size;
                 ^~~~~~~~~~
cc1plus: warning: unrecognized command line option ‘-Wno-deprecated-declaration’
modules/models/CMakeFiles/models.dir/build.make:62: recipe for target 'modules/models/CMakeFiles/models.dir/main.cpp.o' failed
make[2]: *** [modules/models/CMakeFiles/models.dir/main.cpp.o] Error 1
CMakeFiles/Makefile2:1163: recipe for target 'modules/models/CMakeFiles/models.dir/all' failed
make[1]: *** [modules/models/CMakeFiles/models.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2

```

I have already downloaded and installed Eigen specified in `tensorlfow/workspace.bzl` (3.9.0, I have printed the version I extracted from "Eigen/src/Core/util/Macros.h")

However after searching a while, only to find some interfaces like "TensorContractionBlocking" defined in installed eigen3 (/usr/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorContractionBlocking.h) was different from the google version (/usr/local/include/google/tensorflow/third_party/eigen3/unsupported/Eigen/CXX11/src/FixedPoint/MatMatProductAVX2.h)

Here is the captured snapshot from screen:

![Screenshot 2020-04-05 at 6 03 36 PM](https://user-images.githubusercontent.com/61530411/78472569-8fbe2d00-776c-11ea-890c-fbdde9569b42.png)

**Any other info / logs**
See above

**Conclution**

Since the source codes are incorrect, It is impossible to compile it in my cmake project. Any help ?

**Comment by yiakwy-enterprise-roborock** [comment]
@gadagashwini 
I have resolved the problem by manually adding the following path to my FindEigen3.cmake module:

```txt
~  44 # set EIGEN3_INCLUDE_DIR
~  45 # this is where I installed bazel eigen refered by tensorflow 2.2.0rc
~  46 unset(EIGEN3_FOUND)
~  47 # /usr/include/eigen3, ceres depended eigen
~  48 # /usr/local/include/eigen3/, extracted from bazel archive
~  49 # set(EIGEN3_INCLUDE_DIR "/usr/local/include/eigen3")
~  50 set (EIGEN3_INCLUDE_DIR "/home/yiakwy/.cache/bazel/_bazel_yiakwy/729cb3000927cd0e322e439998a82145/external/eigen_archive")
~  51 _eigen3_check_version()
~  52 set(EIGEN3_FOUND ${EIGEN3_VERSION_OK})
```

It seems that the released cmake-3.3.9 used by tensorflow-2.2.0-rc is patched. Only the correct version does not help. Hence I dig into the bazel building system and find the relevant eigen3 files used by tensorflow, it shows that they share template with same number of 5 arguments.

**Comment by gadagashwini** [comment]
@yiakwy-enterprise-roborock, Glad it worked.

**Comment by wwtu** [comment]
@yiakwy-enterprise-roborock
Hi, I encounter exactly the same issue with what you described. Can you tell me how you you solve this problem in details. I didn't catch your points clearly. Thanks.
> @gadagashwini
> I have resolved the problem by manually adding the following path to my FindEigen3.cmake module:
> 
> ```
> ~  44 # set EIGEN3_INCLUDE_DIR
> ~  45 # this is where I installed bazel eigen refered by tensorflow 2.2.0rc
> ~  46 unset(EIGEN3_FOUND)
> ~  47 # /usr/include/eigen3, ceres depended eigen
> ~  48 # /usr/local/include/eigen3/, extracted from bazel archive
> ~  49 # set(EIGEN3_INCLUDE_DIR "/usr/local/include/eigen3")
> ~  50 set (EIGEN3_INCLUDE_DIR "/home/yiakwy/.cache/bazel/_bazel_yiakwy/729cb3000927cd0e322e439998a82145/external/eigen_archive")
> ~  51 _eigen3_check_version()
> ~  52 set(EIGEN3_FOUND ${EIGEN3_VERSION_OK})
> ```
> 
> It seems that the released cmake-3.3.9 used by tensorflow-2.2.0-rc is patched. Only the correct version does not help. Hence I dig into the bazel building system and find the relevant eigen3 files used by tensorflow, it shows that they share template with same number of 5 arguments.
